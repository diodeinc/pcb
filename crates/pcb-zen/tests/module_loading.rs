mod common;
use common::TestProject;

#[test]
#[cfg(not(target_os = "windows"))]
fn snapshot_load_component_via_module() {
    let env = TestProject::new();

    env.add_file(
        "pcb.toml",
        r#"
[module]
name = "test"
"#,
    );

    env.add_file(
        "nested/file/import.zen",
        r#"
def DummyFunction():
    pass
"#,
    );

    env.add_file(
        "C146731.kicad_sym",
        include_str!("resources/C146731.kicad_sym"),
    );

    env.add_file(
        "sub.zen",
        r#"
# Import the component factory from the current directory.
load(".", COMP = "C146731")
load("//nested/file/import.zen", DummyFunction = "DummyFunction")

DummyFunction()

# Instantiate with required pin connections via `pins` dict.
COMP(
    name = "NB3N551DG",
    footprint = "SMD:0805",
    pins = {
        "ICLK": Net("ICLK"),
        "Q1": Net("Q1"),
        "Q2": Net("Q2"),
        "Q3": Net("Q3"),
        "Q4": Net("Q4"),
        "GND": Net("GND"),
        "VDD": Net("VDD"),
        "OE": Net("OE"),
    },
)
"#,
    );

    env.add_file(
        "top.zen",
        r#"
# Import the `sub` module from the current directory and alias it to `Sub`
load(".", Sub = "sub")

Sub(name = "sub")
"#,
    );

    star_snapshot!(env, "top.zen");
}
