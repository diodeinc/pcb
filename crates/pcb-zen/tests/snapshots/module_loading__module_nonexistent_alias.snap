---
source: crates/pcb-zen/tests/module_loading.rs
expression: netlist
---
Error: [TEMP_DIR]test.zen:3:17-49 Failed to resolve module path '@missing/something.zen': File not found: [TEMP_DIR]does_not_exist/something.zen
